### PR TITLE
Fe add category

### PR DESCRIPTION
**Pull request by LukaszBielsko**

### CODE REVIEW

File: frontend/src/authenticated/AddPost.js [hunk 1]
```diff
@@ -52,14 +56,33 @@ const AddPost = props => {
 
   const handleSelect = selectedOptions => {
     if (!selectedOptions) {
+      setUserCategories([]);
```
Comment (MarcinCholewka) [inline frontend/src/authenticated/AddPost.js:59]: Add empty line before return statement :-)

File: frontend/src/authenticated/AddPost.js [hunk 2]
```diff
@@ -52,14 +56,33 @@ const AddPost = props => {
...
+  const handlePublicCheckbox = () => {
+    if (isReviewNeeded && isPublic) {
+      setIsPublic(!isPublic);
+      setIsReviewNeeded(!isReviewNeeded);
+    } else if (isReviewNeeded) {
+      setIsPublic(!isPublic);
+    } else {
+      setIsPublic(!isPublic);
+      setIsReviewNeeded(!isReviewNeeded);
+    }
```
Comment (MarcinCholewka) [inline frontend/src/authenticated/AddPost.js:79]: Create three if statements

File: frontend/src/authenticated/ReviewPost.js [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+  const [hash, setHash] = useState(null);
+
+  useEffect(() => {
+    const fetchpost = async () => {
+      const hash = query.get("hashed_id");
+      const post = await fetchReviewPost(`/api/posts/${hash}/review`);
+      setHash(hash);
+      setPost(post);
+    };
+    fetchpost();
```
Comment (MarcinCholewka) [inline frontend/src/authenticated/ReviewPost.js:24]: add empty line before fetchpost method

File: frontend/src/utils/auth.js [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+};
+
+export const getToken = () => {
+  return window.localStorage.getItem(localStorageKey);
+};
+
+// AUTHENTICATION
+export const isAuthenticated = () => {
+  const token = getToken();
+  if (token) return true;
```
Comment (MarcinCholewka) [inline frontend/src/utils/auth.js:15]: add empty line before if statement
```suggestion
  if (token) {
    return true;
  }
```

File: frontend/src/utils/index.js [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+
+export { fetchUser } from "./dataFetching/fetchUser";
+
+export {
+  convertToSelectOptions,
+  convertReactions,
+  checkHasReacted
+} from "./array/helpers";
+
+export { isAuthenticated, deleteToken } from "./auth";
```
Comment (MarcinCholewka) [inline frontend/src/utils/index.js:28]: ???

**Comment by MarcinCholewka** [comment]
and you need a rebase 😄

File: frontend/src/utils/dataFetching/fetchPosts.js [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+  try {
+    const response = await fetch(url + id);
+    const user = await response.json();
+    const userPosts = user.posts;
+
+    return userPosts;
+  } catch (error) {
+    console.warn(error);
+  }
+};
```
Comment (Wozniaxos) [inline frontend/src/utils/dataFetching/fetchPosts.js:41]: Hmm... why fetch posts is in other scope than post crud? CRUD stands for Create Read - which is fetch, update and delete.

File: frontend/src/utils/dataFetching/CRUDposts.js [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+  const response = await fetch(url, {
+    method: "DELETE",
+    headers: {
+      Authorization: `Bearer ${getToken()}`,
+      "Content-Type": "application/json"
+    }
+  });
+
+  return response.ok;
+};
```
Comment (Wozniaxos) [inline frontend/src/utils/dataFetching/CRUDposts.js:42]: Why CRUD is in scope of dataFetching - since is not only fetching but also modifying

Comment (Wozniaxos) [reply frontend/src/utils/dataFetching/CRUDposts.js:42]: Btw you can have here something like 
```js
export cont async request(action, url, body) => {
  return fetch(url, {
    method: method,
    headers: {
      ....
    }
  })
}

//then

export const deleteData = async (url, data) => {
  const response = await request("DELETE", url, data)
  return response.ok
}

```
and all of other methods

File: frontend/src/utils/dataFetching/fetchPosts.js (hunk 6 above)
Comment (LukaszBielsko) [reply frontend/src/utils/dataFetching/fetchPosts.js:41]: makes perfect sense

**Pull request merged**